binstar_client/utils/detect.py

```diff
@@ -22,13 +23,22 @@ def file_handler(filename, fileobj, *args, **kwargs):
     return ({}, {'description': ''},
             {'basename': path.basename(filename), 'attrs':{}})
 
-detectors = {'conda':inspect_conda_package,
-             'pypi': inspect_pypi_package,
-             'r': inspect_r_package,
-             'ipynb': inspect_ipynb_package,
-             conda_installer.PACKAGE_TYPE: conda_installer.inspect_package,
-             'file': file_handler,
-             }
+detectors = {
+    'conda': inspect_conda_package,
+    'pypi': inspect_pypi_package,
+    'r': inspect_r_package,
+    'ipynb': inspect_ipynb_package,
+    'env': inspect_env_package,
+    conda_installer.PACKAGE_TYPE: conda_installer.inspect_package,
+    'file': file_handler,
+}
+
+
+def is_environment(filename):
+    log.debug("Testing if environment file ..")
+    if filename.endswith('.yml') or filename.endswith('.yaml'):
+        return True
```

Comment:
I thought about it, I think we will end up doing that later once we try to upload `anaconda-cloud` definitions or something like that. But I don't see any reason why we should do that now